src/DssSpell.t.sol

```diff
@@ -1240,4 +1439,344 @@ contract DssSpellTest is DssSpellTestBase {
+                VestedRewardsDistributionJobLike(CRON_REWARDS_DIST_JOB).work(network, args);
+            }
+        } while(isWorkable);
+
+        // Verify both distributions have been executed
+        assertTrue(foundUsdsSpk, "USDS farm not distributed");
+        assertTrue(foundLsskySpk, "LSSKY farm not distributed");
+
+        // Verify both farms have reward rates > 0
+        assertGt(StakingRewardsLike(REWARDS_USDS_SPK).rewardRate(), 0, "USDS farm reward rate still zero");
+        assertGt(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardRate(), 0, "LSSKY farm reward rate still zero");
+
+        // Check if the job is no longer workable
+        // Note: `workable` is not a view function in this case, so we need to revert to the previous state after calling it.
+        uint256 beforeThirdWorkable = vm.snapshotState();
+        (bool thirdWorkable, ) = VestedRewardsDistributionJobLike(CRON_REWARDS_DIST_JOB).workable(network);
+        vm.revertToStateAndDelete(beforeThirdWorkable);
+
+        assertFalse(thirdWorkable, "VestedRewardsDistributionJob should not be workable");
+    }
+
+    function test_treasuryMkrToSkyConversion_leftoverMkr() public {
+        // Get the unpaid MKR for vest ids 9, 18, 24, 35, 37, and 39
+        VestAbstract vestMkrTreasury = VestAbstract(addr.addr("MCD_VEST_MKR_TREASURY"));
+        uint256 unpaidMkr = vestMkrTreasury.unpaid(9) +
+            vestMkrTreasury.unpaid(18) +
+            vestMkrTreasury.unpaid(24) +
+            vestMkrTreasury.unpaid(35) +
+            vestMkrTreasury.unpaid(37) +
+            vestMkrTreasury.unpaid(39);
+
+        // Cast the spell
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "TestError/spell-not-done");
+
+        // Check that the MKR balance has decreased by the expected amount
+        uint256 mkrBalanceAfter = mkr.balanceOf(address(pauseProxy));
+
+        // Verify MKR was converted to SKY
+        assertEq(mkrBalanceAfter, unpaidMkr, "MKR balance should equal unpaid vest amount");
+    }
+
+    function test_disableMkrSkyLegacyConverter() public {
+        // Check if MKR_SKY_LEGACY exists in the chainlog before the spell
+        try chainLog.getAddress("MKR_SKY_LEGACY") returns (address mkrSkyLegacy) {
+            // Get SKY token from the MKR_SKY_LEGACY converter
+            address skyToken = MkrSkyLike(mkrSkyLegacy).sky();
+
+            // Check if MKR_SKY_LEGACY has authority on SKY token before the spell
+            assertEq(WardsLike(skyToken).wards(mkrSkyLegacy), 1, "MKR_SKY_LEGACY should have authority on SKY before spell");
+
+            // Cast the spell
+            _vote(address(spell));
+            _scheduleWaitAndCast(address(spell));
+            assertTrue(spell.done(), "TestError/spell-not-done");
+
+            // Check if MKR_SKY_LEGACY has been removed from the chainlog
+            try chainLog.getAddress("MKR_SKY_LEGACY") {
```

Comment:
Could a `vm.expectRevert` be used here? Is less verbose than a try-catch

Reply:
Fixed in [a0d395c](https://github.com/sky-ecosystem/spells-mainnet/pull/479/commits/a0d395cd8899e1dfd28eddb37d4c4fa38a7be4c6)